src/util.c

```diff
@@ -364,3 +364,65 @@ debug_hex_dump_slice(platform_log_handle *plh, uint64 grouping, slice data)
 {
    debug_hex_dump(plh, grouping, slice_length(data), slice_data(data));
 }
+
+/*
+ * Format a size value with unit-specifiers, in an output buffer.
+ * Returns 'outbuf', as ptr to size-value snprintf()'ed as a string.
+ */
+char *
+size_to_str(char *outbuf, size_t outbuflen, size_t size)
+{
+   debug_assert(outbuflen >= SIZE_TO_STR_MINLEN, "outbuflen=%lu\n", outbuflen);
+   size_t unit_val  = 0;
+   size_t frac_val  = 0;
+   bool   is_approx = FALSE;
+
+   char  fmtstr[SIZE_TO_STR_LEN];
+   char *units = NULL;
+   if (size >= TiB) {
+      unit_val  = B_TO_TiB(size);
+      frac_val  = B_TO_TiB_FRACT(size);
+      is_approx = (size > TiB_TO_B(unit_val));
+      units     = "TiB";
+
+   } else if (size >= GiB) {
+      unit_val  = B_TO_GiB(size);
+      frac_val  = B_TO_GiB_FRACT(size);
+      is_approx = (size > GiB_TO_B(unit_val));
+      units     = "GiB";
+
+   } else if (size >= MiB) {
+      unit_val  = B_TO_MiB(size);
+      frac_val  = B_TO_MiB_FRACT(size);
+      is_approx = (size > MiB_TO_B(unit_val));
+      units     = "MiB";
+
+   } else if (size >= KiB) {
+      unit_val  = B_TO_KiB(size);
+      frac_val  = B_TO_KiB_FRACT(size);
+      is_approx = (size > KiB_TO_B(unit_val));
+      units     = "KiB";
+   } else {
+      unit_val = size;
+      units    = "bytes";
+   }
+   if (frac_val || is_approx) {
+      snprintf(fmtstr, sizeof(fmtstr), "~%%ld.%%d %s", units);
+   } else {
+      snprintf(fmtstr, sizeof(fmtstr), "%%ld %s", units);
+   }
+   snprintf(outbuf, outbuflen, fmtstr, unit_val, frac_val);
+   return outbuf;
+}
+
+/*
+ * Sibling of size_to_str(), but uses user-provided print format specifier.
+ * 'fmtstr' is expected to have just one '%s', and whatever other text user
+ * wishes to print with the output string.
+ */
+char *
```

Comment:
I don't see a reason for this function to exist.  First, I don't like the call sites.  They are working too hard for a minor change in the output.

Second, if we do want to keep the unnecessarily complicated code, then call sites should just do the snprintf directly.

Reply:
The problem as I explained previously, Rob, is that we want to see the formatted string enclosed in `( )` and the outer tabular output has all these columnar alignments. That's why the calls in the call-sites are working a bit harder to conform to this aligned output.

Earlier, in my dev, I actually had this as `size_to_str2()` which would only support one output formatting; i.e. enclosing the output in `(%s)`. That would work as well.

I just thought it would be better to make this formatting a bit more general, which is why I changed this function to `size_to_fmtstr()`, receiving a user-provided format string, which in all cases (currently) is simply `(%s)`. 
It also makes the call self-documenting in the caller's context to see that the output will be enclosed in `()`.

I have thought about this quite a bit and have fuddled with ways to try to get the printing to work even while retaining the aesthetics of the aligned columnar output. I couldn't think of any better scheme.

Honestly, I don't think the existence of this sibling printing routine is a huge annoyance. In fact, the work of `size_to_fmtstr()` has been even improved by the introduction of `size_str()` macro-ization you suggested.

So, I think it's an improvement overall! (I'd like to retain this.)